models/items.js

```diff
@@ -114,7 +118,7 @@ class Items {
     const searchObj = {}
     properties.forEach(property => {
       const column = datasetTypeList.find(column => column.name === property)
-      if (searchQuery[property] !== '') {
+      if (searchQuery[property] !== '' && column !== undefined) {
```

Comment:
Rather than checking for undefined, if we check for it being `True` then it will have a better chance of passing through try block.